crates/bevy_core_pipeline/src/deferred/node.rs

```diff
@@ -0,0 +1,192 @@
+    view::ViewDepthTexture,
+};
+#[cfg(feature = "trace")]
+use bevy_utils::tracing::info_span;
+
+use crate::core_3d::{Camera3d, Camera3dDepthLoadOp};
+use crate::prepass::{DepthPrepass, MotionVectorPrepass, NormalPrepass, ViewPrepassTextures};
+
+use super::{AlphaMask3dDeferred, Opaque3dDeferred};
+
+/// Render node used by the prepass.
+///
+/// By default, inserted before the main pass in the render graph.
+#[derive(Default)]
+pub struct DeferredNode;
+
+impl ViewNode for DeferredNode {
+    type ViewQuery = (
+        &'static ExtractedCamera,
+        &'static RenderPhase<Opaque3dDeferred>,
+        &'static RenderPhase<AlphaMask3dDeferred>,
+        &'static ViewDepthTexture,
+        &'static ViewPrepassTextures,
+        &'static Camera3d,
+        Option<&'static DepthPrepass>,
+        Option<&'static NormalPrepass>,
+        Option<&'static MotionVectorPrepass>,
+    );
+
+    fn run(
+        &self,
+        graph: &mut RenderGraphContext,
+        render_context: &mut RenderContext,
+        (
+            camera,
+            opaque_deferred_phase,
+            alpha_mask_deferred_phase,
+            view_depth_texture,
+            view_prepass_textures,
+            camera_3d,
+            depth_prepass,
+            normal_prepass,
+            motion_vector_prepass,
+        ): QueryItem<Self::ViewQuery>,
+        world: &World,
+    ) -> Result<(), NodeRunError> {
+        let view_entity = graph.view_entity();
+
+        if let Some(msaa) = world.get_resource::<Msaa>() {
+            match msaa {
+                Msaa::Off => (),
+                _ => panic!("MSAA not supported when using deferred rendering."),
+            }
+        }
+
+        let mut color_attachments = vec![];
+        color_attachments.push(
+            view_prepass_textures
+                .normal
+                .as_ref()
+                .map(|view_normals_texture| RenderPassColorAttachment {
+                    view: &view_normals_texture.default_view,
+                    resolve_target: None,
+                    ops: Operations {
+                        load: if normal_prepass.is_some() {
+                            LoadOp::Load // load if the normal_prepass has run
+                        } else {
+                            LoadOp::Clear(Color::BLACK.into())
+                        },
+                        store: true,
+                    },
+                }),
+        );
+        color_attachments.push(view_prepass_textures.motion_vectors.as_ref().map(
+            |view_motion_vectors_texture| RenderPassColorAttachment {
+                view: &view_motion_vectors_texture.default_view,
+                resolve_target: None,
+                ops: Operations {
+                    load: if motion_vector_prepass.is_some() {
+                        LoadOp::Load // load if the motion_vector_prepass has run
+                    } else {
+                        LoadOp::Clear(Color::rgb_linear(0.0, 0.0, 0.0).into())
```

Comment:
Why?

Reply:
If `motion_vector_prepass` has already run `LoadOp::Clear` already happened there and we don't want to clear the motion vectors from that pass earlier in the same frame.

Reply:
Oh, sorry wasn't very clear here. I mainly meant why not just use the Color::BLACK constant like you used above it.

Reply:
Ahh, gotcha. No reason. I'll swap it out.